src/js/media-captions-list.js

```diff
@@ -0,0 +1,137 @@
+
+
+class MediaCaptionsList extends MediaChromeListbox {
+  #subs = [];
+  #caps = [];
+
+  static get observedAttributes() {
+    return [
+      ...super.observedAttributes,
+      MediaUIAttributes.MEDIA_CAPTIONS_LIST,
+      MediaUIAttributes.MEDIA_CAPTIONS_SHOWING,
+      MediaUIAttributes.MEDIA_SUBTITLES_LIST,
+      MediaUIAttributes.MEDIA_SUBTITLES_SHOWING,
+    ];
+  }
+
+  attributeChangedCallback(attrName, oldValue, newValue) {
+    if (attrName === MediaUIAttributes.MEDIA_SUBTITLES_LIST && oldValue !== newValue) {
+      this.#subs = parseTextTracksStr(this.getAttribute(MediaUIAttributes.MEDIA_SUBTITLES_LIST) ?? '');
+      this.#render();
+
+    } else if (attrName === MediaUIAttributes.MEDIA_CAPTIONS_LIST && oldValue !== newValue) {
+      this.#caps = parseTextTracksStr(this.getAttribute(MediaUIAttributes.MEDIA_CAPTIONS_LIST) ?? '');
+      this.#render();
+
+    } else if (attrName === MediaUIAttributes.MEDIA_SUBTITLES_SHOWING && oldValue !== newValue) {
+      const selectedTrack = parseTextTrackStr(this.getAttribute(MediaUIAttributes.MEDIA_SUBTITLES_SHOWING) ?? '');
+
+      this.#subs.forEach(track => {
+        track.selected = track.language === selectedTrack.language && track.label === selectedTrack.label;
+      });
+      this.#render();
+
+    } else if (attrName === MediaUIAttributes.MEDIA_CAPTIONS_SHOWING && oldValue !== newValue) {
+      const selectedTrack = parseTextTrackStr(this.getAttribute(MediaUIAttributes.MEDIA_CAPTIONS_SHOWING) ?? '');
+
+      this.#caps.forEach(track => {
+        track.selected = track.language === selectedTrack.language && track.label === selectedTrack.label;
+      });
+      this.#render();
+    }
+
+    super.attributeChangedCallback(attrName, oldValue, newValue);
+  }
+
+  connectedCallback() {
+    this.#render();
+
+    this.addEventListener('change', this.#onChange);
+
+    super.connectedCallback();
+  }
+
+  disconnectedCallback() {
+    this.removeEventListener('change', this.#onChange);
+
+    super.disconnectedCallback();
+  }
+
+  #perTypeRender(tracks, type) {
+    tracks.forEach(track => {
+      let option = track.el;
+      let alreadyInDom = true;
+
+      if (!option) {
+        option = document.createElement('media-chrome-listitem');
+        alreadyInDom = false;
+
+        option.value = type + '!' + formatTextTrackObj(track);
+        option.innerHTML = track.label + (track.kind === 'captions' ? ' (cc)' : '');
+      }
+
+
+      if (track.selected) {
+        option.setAttribute('aria-selected', 'true');
+      } else {
+        option.setAttribute('aria-selected', 'false');
+      }
+
+      if (!alreadyInDom) {
+        this.append(option);
+        track.el = option;
+      }
+    });
+  }
+
+  #render() {
+    this.#perTypeRender(this.#subs, 'subs');
+    this.#perTypeRender(this.#caps, 'cc');
+  }
+
+  #onChange(e) {
+    const [newType, selectedOption] = this.selectedOptions[0]?.value?.split('!') ?? [];
```

Comment:
suggestion(non-blocking): given that we only use `value` for identifying data type here and it's "meaningless" to the base `media-chrome-list-item`, we should consider just creating an instance-level `Map` on `media-captions-list` where keys are the "options" (elements) and values are the `value`.

Reply:
Since we're keeping it on listitem, we don't need this